salesagility/SuiteCRM
# Fixed #8989 - Removed legacy code for export from modules

This pull request removes the obsolete legacy method create_export_query() from modules while making sure non-primary email addresses are still exported.

## Description
For many modules the SugarBean::create_export_query() method is overwritten with a derived method create_export_query() containing legacy code. The legacy code used to be useful for adding assigned user, email address and other related data.

This functionality has already been implemented in SugarBean::create_new_list_query(), which makes these methods obsolete. Moreover, they can't handle exports where a filter on a relate field is used, see https://github.com/salesagility/SuiteCRM/issues/8989

## Motivation and Context
This pull request removes obsolete create_export_query() methods. Where these methods applied filtering and ordering, this has been implemented by calls to SugarBean::create_export_query() or SugarBean::create_new_list_query()

Some modules also exported the non-primary email addresses by adding a field 'email_addresses_non_primary', which was picked up by include/export_utils.php:export(). To make sure non-primary emails are still exported, the export() method now always exports these when the field 'email1' is included in the export result and when the module supports non-primary email addresses

## How To Test This
See: https://github.com/salesagility/SuiteCRM/issues/8989#issuecomment-742767434

Especially test the export of the modules affected (I already did).

## Types of changes


### Final checklist
